core/src/main/java/com/linecorp/armeria/server/docs/DocService.java

```diff
@@ -313,12 +315,13 @@ private CompletableFuture<AggregatedHttpFile> loadVersions(Executor executor) {
             }, executor));
         }
 
-        private CompletableFuture<ServiceSpecification> generateServiceSpecification(Executor executor) {
+        private CompletableFuture<ServiceSpecification> generateServiceSpecification(Executor executor,
+                                                                                     Route docServiceRoute) {
             return CompletableFuture.supplyAsync(() -> {
                 final DocStringSupport docStringSupport = new DocStringSupport(services);
                 ServiceSpecification spec = generate(services);
                 spec = docStringSupport.addDocStrings(spec);
-                spec = exampleSupport.addExamples(spec);
+                spec = exampleSupport.addExamples(spec, docServiceRoute);
```

Comment:
Technically, `docServiceRoute` isn't related to the example. 
How about specifying it when calling `generate` method above?
```java
ServiceSpecification spec = generate(services, docServiceRoute);

// generate method
private ServiceSpecification generate(List<ServiceConfig> services, Route docServiceRoute) {
    return ServiceSpecification.merge(
            plugins.stream()
                   .map(plugin -> plugin.generateSpecification(
                           findSupportedServices(plugin, services),
                           filter, descriptiveTypeInfoProvider))
                   .collect(toImmutableList()), docServiceRoute);
}

// merge method
public static ServiceSpecification merge(Iterable<ServiceSpecification> specs, Route docServiceRoute) {
    return new ServiceSpecification(
            Streams.stream(specs).flatMap(s -> s.services().stream())::iterator,
            Streams.stream(specs).flatMap(s -> s.enums().stream())::iterator,
            Streams.stream(specs).flatMap(s -> s.structs().stream())::iterator,
            Streams.stream(specs).flatMap(s -> s.exceptions().stream())::iterator,
            ImmutableList.of(), docServiceRoute);
}
```